File: natura-server/src/main/java/com/natura/web/server/entities/Entry.java
package com.natura.web.server.entities;

import jakarta.persistence.Column;
import jakarta.persistence.DiscriminatorColumn;
import jakarta.persistence.DiscriminatorType;
import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Inheritance;
import jakarta.persistence.InheritanceType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import java.util.Date;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
@EqualsAndHashCode(callSuper = true)
@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "DISCRIMINATOR", discriminatorType = DiscriminatorType.STRING)
@DiscriminatorValue("ENTRY")
public abstract class Entry extends ValidableItem {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String name;

  @ManyToOne(fetch = FetchType.EAGER)
  @JoinColumn(name = "user_id")
  private User createdBy;

  @OneToOne
  @JoinColumn(name = "image_id")
  Image image;

  @Column(columnDefinition = "TEXT")
  private String description;

  private String location;

  private Date date;

  protected Entry() {
    super();
  }

  protected Entry(String name) {
    this();
    this.name = name;
  }

  protected Entry(String name, Date date, String description, String location) {
    this();
    this.name = name;
    this.date = date;
    this.description = description;
    this.location = location;
  }
}

File: natura-go-backend/src/natura/models/entry.go
package models

import(
	"time"
)

type Entry struct {
	Id        		int64  	`gorm:"primary_key, AUTO_INCREMENT" json:"id"`
	Name	  		string	`json:"name"`
	Description		string	`gorm:"type:TEXT" json:"description"`
	Location  		string	`json:"location"`
	IsValidated		bool	`gorm:"column:validated, type=TINYINT(1)" json:"validated"` 
	Date			time.Time `json:"date"`
	CreatedByID		int64	`gorm:"column:user_id"`
	CreatedBy  		User	`json:"createdBy"`
	ImageID			int64 	`gorm:"column:image_id"`
	Image			Image	`json:"image"`
	Discriminator	SpeciesType

}

func (b *Entry) TableName() string {
	return "entry"
}